src/output-json-anomaly.c

```diff
@@ -328,6 +445,11 @@ void JsonAnomalyLogRegister (void)
         "eve-log.anomaly", JsonAnomalyLogInitCtxSub, JsonAnomalyLogger,
         JsonAnomalyLogCondition, JsonAnomalyLogThreadInit, JsonAnomalyLogThreadDeinit,
         NULL);
+
+    OutputRegisterTxSubModule(LOGGER_JSON_ANOMALY, "eve-log", MODULE_NAME,
```

Comment:
hmm how often is JsonAnomalyTxLogger called per tx? I fear only once, probably from the first packet for that tx?

Reply:
I've modified the output process -- for a given transaction, if any logger has deferred logging the transaction, the "wild card" logger will also defer. This relies on the anomaly logger being at the end of the logging chain (in my current implementation). 

This ensures that the transaction is not examined by the anomaly tx logger until the transaction has been deemed complete.